src/Octoshift/GithubApi.cs

```diff
@@ -459,5 +460,145 @@ public virtual async Task<string> GetArchiveMigrationUrl(string org, int migrati
             var response = await _client.GetNonSuccessAsync(url, HttpStatusCode.Found);
             return response;
         }
+
+        public virtual async Task<Mannequin> GetMannequin(string orgid, string username)
+        {
+            var url = $"{_apiUrl}/graphql";
+
+            var payload = BuildListMannequinPayload(orgid, null);
+
+            JObject data;
+            do
+            {
+                var response = await _client.PostAsync(url, payload);
+                data = JObject.Parse(response);
+
+                var mannequin = data["data"]["node"]["mannequins"]["nodes"].FirstOrDefault(node =>
+                    { return username.Equals((string)node["login"], System.StringComparison.OrdinalIgnoreCase); }
+                );
+
+                if (mannequin != null)
+                {
+                    Claimant claimant = null;
+
+                    if (mannequin["claimant"].Any())
+                    {
+                        claimant = new Claimant
+                        {
+                            Id = (string)mannequin["claimant"]["id"],
+                            Login = (string)mannequin["claimant"]["login"]
+                        };
+                    }
+
+                    return new Mannequin
+                    {
+                        Id = (string)mannequin["id"],
+                        Login = (string)mannequin["login"],
+                        MappedUser = claimant
+
+                    };
+                }
+
+                payload = BuildListMannequinPayload(orgid, (string)data["data"]["node"]["mannequins"]["pageInfo"]["endCursor"]);
+
+            } while ((bool)data["data"]["node"]["mannequins"]["pageInfo"]["hasNextPage"]);
```

Comment:
I am working on a method that supports getting paginated GQL results, the final PR should be out today (if no other urgent thing comes in the way). I suggest holding off on merging this PR and re-write this method using the GQL paginated method.  It should be a fairly simple rewrite, all the pagination logic will be handled in the method 😃

Reply:
#264 Right?

This particular method does lazy pagination, I did a cursory look at the PR and I don't think lazy loading is a supported pattern.  Did I got that right?